tools/agentlint/internal/rivertags/rivertags.go

```diff
@@ -63,7 +78,22 @@ func run(p *analysis.Pass) (interface{}, error) {
 			}
 
 			matches := riverTagRegex.FindAllStringSubmatch(s.Tag(i), -1)
-			if len(matches) == 0 {
+			if len(matches) == 0 && hasRiverTags {
+				// If this struct has River tags, but this field only has json/yaml
+				// tags, emit an error.
+				jsonMatches := jsonTagRegex.FindAllStringSubmatch(s.Tag(i), -1)
+				yamlMatches := yamlTagRegex.FindAllStringSubmatch(s.Tag(i), -1)
+
+				if len(jsonMatches) != 0 || len(yamlMatches) != 0 {
```

Comment:
a bit confusing logic. Would it be better to say `if len(jsonMatches) + len(yamlMatches) > 0` ? Maybe even just changing the `!=`s to `>` would make it read easier.

Reply:
I think this may be personal taste. I prefer the logical or statement because I read it literally as "if there are any JSON matches, or if there are any YAML matches." Changing it to a sum is a bit weirder to me. 

I can definitely change to `len(jsonMatches) > 0 || len(yamlMatches) > 0` though.